packages/flutter/lib/src/widgets/spell_check.dart

```diff
@@ -30,11 +31,19 @@ class SpellCheckConfiguration {
     :  _spellCheckEnabled = false,
        spellCheckService = null,
        spellCheckSuggestionsToolbarBuilder = null,
-       misspelledTextStyle = null;
+       misspelledTextStyle = null,
+       misspelledSelectionColor = null;
 
   /// The service used to fetch spell check results for text input.
   final SpellCheckService? spellCheckService;
 
+  /// The color the paint the selection highlight when spell check is showing
+  /// suggestions for a misspelled word.
+  ///
+  /// For example, on iOS, the selection appears red while the spell check menu
+  /// is showing.
+  final Color? misspelledSelectionColor;
```

Comment:
I considered the alternative of not making this configurable, but with the way we create `SpellCheckConfiguration`s in the text fields, I think this makes the most sense :)

Reply:
Ok cool, this was one of the main decisions I wanted to get your feedback on.  I think that however we do it, CupertinoTextField needs to tell EditableText what the color should be somehow, so we can't make it completely private.  The other alternative I considered was adding a parameter `spellCheckSelectionColor` to EditableText, but I think this way seems more organized.

Reply:
Actually before I commit 100% to this, @MitchellGoodwin do you have any thoughts?

So currently TextField and CupertinoTextField read the selection color from DefaultSelectionStyle and then pass it to EditableText as the selectionColor parameter.  This PR adds the ability for CupertinoTextField to change the selection to red for spell check by passing the new misspelledSelectionColor value to EditableText as a part of SpellCheckConfiguration.

The question is, does this align with your vision for theming/styles in Cupertino?

Reply:
Yes, I don't see any issues with that. 👍

Reply:
Thanks!